test/statetest/statetest.cpp

```diff
@@ -16,31 +16,42 @@ namespace
 class StateTest : public testing::Test
 {
     fs::path m_json_test_file;
+    std::optional<std::string> m_test_name;
```

Comment:
This is rather a filter than a name. Maybe name it `m_name_filter` or just `m_filter`.